pyriemann/estimation.py

```diff
@@ -704,6 +705,107 @@ def transform(self, X):
         return covmats
 
 
+###############################################################################
+
+
+class Kernels(BaseEstimator, TransformerMixin):
+    """Estimation of kernel matrix between channels of time series.
+
+    Perform a kernel matrix estimation for each given time series, evaluating a
+    kernel function between each pair of channels (rather than between pairs of
+    time samples) and allowing to extract nonlinear channel relationship [1]_.
+
+    For an input time series :math:`X \in \mathbb{R}^{c \times t}`, kernel
+    function :math:`\kappa()` is computed between channels :math:`i` and
+    :math:`j`:
+
+    .. math::
+        K_{i,j} = \kappa \left( X[i], X[j] \right)
+
+    Linear kernel is related to :class:`pyriemann.estimation.Covariances` [1]_,
+    but this class allows to generalize to nonlinear relationships.
+
+    Parameters
+    ----------
+    metric : string, default='linear'
+        The metric to use when computing kernel function between channels [2]_:
+        'linear', 'poly', 'polynomial', 'rbf', 'laplacian', 'cosine'.
+    n_jobs : int, default=None
+        The number of jobs to use for the computation [2]_. This works by
+        breaking down the pairwise matrix into n_jobs even slices and computing
+        them in parallel.
+
+    See Also
+    --------
+    Covariances
+
+    Notes
+    -----
+    .. versionadded:: 0.3.1
+
+    References
+    ----------
+    .. [1] `Beyond Covariance: Feature Representation with Nonlinear Kernel
+        Matrices
+        <https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Wang_Beyond_Covariance_Feature_ICCV_2015_paper.pdf>`_
+        L. Wang, J. Zhang, L. Zhou, C. Tang, W Li. ICCV, 2015.
+    .. [2]
+        https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_kernels.html
+    """  # noqa
+
+    def __init__(self, metric='linear', n_jobs=None):
+        """Init."""
+        self.metric = metric
+        self.n_jobs = n_jobs
```

Comment:
some kernels have parameters eg the gamma in RBF. You should expose this option from the **kwds in the sklearn.metrics.pairwise.pairwise_kernels function

Reply:
yep, good idea!